Add LMS app template for fine-grained authorization

Tabintel opened the pull request:
# Pull Request

## Description

Adds a new Terraform template `lms-app` for a Learning Management System (LMS) with resources (`course`, `enrollment`, `assignment`), roles (`student`, `teacher`, `teaching_assistant`, `admin`), attributes, relationships, and fine-grained policies (e.g., students can read enrolled courses). Includes a `README.md` with usage instructions. The template is designed to work with a [Go-based LMS using Appwrite](https://github.com/Tabintel/appwrite_permit_lms) for authentication and data storage, as described in the accompanying article.

## Type of Change

- [ ] Bug fix
- [ ] New feature/command
- [x] Documentation update
- [ ] Code refactoring
- [ ] Performance improvement
- [ ] Test addition/update
- [ ] Other (please describe):

## Checklist

- [ ] I have created an issue and linked it in this PR
- [x] I have created a branch (add-lms-app-template) from `main` with an appropriate name (e.g., `fix/issue-123`, `feature/new-command`)
- [x] My code follows the project's coding style guidelines
- [ ] I have added tests for my changes (>90% coverage of new code)
- [x] I have updated the documentation if necessary
- [x] All tests pass locally
- [x] Lint checks pass locally
- [x] I have reviewed my own code for potential issues

## New Command Details (if applicable)

- [ ] Command is placed in the `src/commands` directory
- [ ] Command file contains only argument configuration and a root command component
- [ ] Command is wrapped with the `AuthProvider` component
- [ ] Command has an optional `apiKey` argument
- [ ] API key scope is declared for the command
- [ ] Documentation added to the README

## Additional Notes

The `lmsapp.tf` file was initially ignored by .gitignore (likely due to a `*.tf` rule). I used `git add -f `to include it, as Terraform templates are expected in `source/templates/` (example: `blogging-platform.tf`). _Please advise if there is a preferred approach._

## Screenshots/Recordings

Here is a screenshot of the `lmsapp` folder added to the `source/templates` directory

<img width="894" height="253" alt="image" src="https://github.com/user-attachments/assets/8a93b231-6656-40e0-ab1d-fce9620ea592" />

gemanor commented on source/templates/lms-app/lmsapp.tf:
```diff
@@ -0,0 +1,189 @@
+terraform {
```
Please move it to the `source/template` folder, and add the template description to the root `readme` file
Also, revert the .gitignore

gemanor commented on source/templates/lms-app/lmsapp.tf:
```diff
@@ -0,0 +1,189 @@
+  required_providers {
+    permitio = {
+      source  = "permitio/permit-io"
+      version = "~> 0.0.12"
```
Should be same as other templates

gemanor reviewed (changes requested):
Please address the comments and attach screenshot of environment where you imported the policy

gemanor commented:
Hey @Tabintel - can you address the comments and solve the failing build?

Tabintel commented:
> Please address the comments and attach screenshot of environment where you imported the policy

Hi @gemanor, I have addressed the comments. Here is the screenshot
![lmsapp](https://github.com/user-attachments/assets/33cae383-3918-480c-9c1b-3b1d7c040133)
